plugins/outputs/azure_monitor/azure_monitor.go

```diff
@@ -359,12 +359,19 @@ func translate(m telegraf.Metric, prefix string) (*azureMonitorMetric, error) {
 			continue
 		}
 
-		if tag.Key == "" || tag.Value == "" {
+		var key = tag.Key
+		var value = tag.Value
+
+		if key == "" {
 			continue
 		}
 
-		dimensionNames = append(dimensionNames, tag.Key)
-		dimensionValues = append(dimensionValues, tag.Value)
+		if value == "" {
+			value = "<empty>"
```

Comment:
Can dimensions be added with the value as the empty string instead of using a placeholder value?

Reply:
If we can't have empty strings at all, I'd like to try this instead:
```diff
diff --git a/plugins/outputs/azure_monitor/azure_monitor.go b/plugins/outputs/azure_monitor/azure_monitor.go
index 5c435ac0..1c933072 100644
--- a/plugins/outputs/azure_monitor/azure_monitor.go
+++ b/plugins/outputs/azure_monitor/azure_monitor.go
@@ -340,6 +340,9 @@ func hashIDWithTagKeysOnly(m telegraf.Metric) uint64 {
        h.Write([]byte(m.Name()))
        h.Write([]byte("\n"))
        for _, tag := range m.TagList() {
+               if tag.Key == "" || tag.Value == "" {
+                       continue
+               }
                h.Write([]byte(tag.Key))
                h.Write([]byte("\n"))
        }
```

Reply:
Yes. This is better implementation. I've tested here with success.
![image](https://user-images.githubusercontent.com/815086/49845808-64f9d980-fdaf-11e8-9ba3-919bdbac8551.png)